### PR TITLE
Added arm64 architecture support in python-build for macOS 

### PR DESCRIPTION
**Pull request by anton-petrov**
Make sure you have checked all steps below.

### Prerequisite
* [ ] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
* [ ] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
* [ ] My PR addresses the following pyenv issue (if any)
  - https://github.com/pyenv/pyenv/issues/XXXX

### Description
- [ ] Here are some details about my PR

### Tests
- [ ] My PR adds the following unit tests (if any)

### CODE REVIEW

File: plugins/python-build/bin/python-build [hunk 1]
```diff
@@ -995,7 +995,13 @@ build_package_activepython() {
 
 anaconda_architecture() {
   case "$(uname -s)" in
-  "Darwin" ) echo "MacOSX-x86_64" ;;
+  "Darwin" )
+    case "$(uname -m)" in
+    "arm64" ) echo "MacOSX-arm64" ;;
```
Comment (chrahunt) [inline plugins/python-build/bin/python-build:1000]: Is it safe to assume that macOS users on ARM always want the ARM version? If some dependencies do not support ARM, they may prefer x86_64 (executed via Rosetta).

File: plugins/python-build/share/python-build/anaconda3-2020.11 [hunk 2]
```diff
@@ -0,0 +1,19 @@
+case "$(anaconda_architecture 2>/dev/null || true)" in
```
Comment (chrahunt) [inline plugins/python-build/share/python-build/anaconda3-2020.11:1]: This looks unrelated to the arm64 change. If we pull this file into it's own PR then it is easier to review. Apologies, I did not communicating this well on #1766.

File: plugins/python-build/bin/python-build (hunk 1 above)
Comment (anton-petrov) [reply plugins/python-build/bin/python-build:1000]: Hi @chrahunt, the approach I suggested is safe and here's why: macOS users on ARM can run programs in x86_64  mode on Rosetta 2 translator quite simple and transparent with the command `arch x86_64`. For example, `arch x86_64 pyenv install 3.9.1` build and install python for x86_64 architecture. In this way, the "arch" command runs commands under Rosetta 2. This approach is checked and tested on my MacBook Air M1 - everything works well! In fact, this is the only (correct) approach, that supported by Apple, and, may be, it would be incorrect to add any new parameters for pyenv, such as `pyenv install 3.9.1 x86_64`ю

Comment (chrahunt) [reply plugins/python-build/bin/python-build:1000]: Thank you for the explanation! I agree, in that light this seems like the right approach.

File: plugins/python-build/bin/python-build [hunk 3]
```diff
@@ -995,7 +995,13 @@ build_package_activepython() {
...
+    "i386" | "x86_64" ) echo "MacOSX-x86_64" ;;
+    * ) return 1 ;;
```
Comment (chrahunt) [inline plugins/python-build/bin/python-build:1002]:
```suggestion
    * ) echo "MacOSX-x86_64" ;;
```

to preserve the original behavior - it may be unlikely that someone is relying on it, but it's less work for now to not ask the question.
